### DISCUSSION TITLE
Vaultwarden Docker with full disk unauthenticates me from all sessions and causes me to lose account access

### DISCUSSION THREAD

**Discussion by TobiasS1402** [dani-garcia/vaultwarden#2200] [Q&A]
### Subject of the issue
Full disk on Debian 10 causes me to get unauthenticated on any session and lose access to my account

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.22.2
* Web-vault version: v2.21.1
* Running within Docker: true
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (CF-Connecting-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.35.4
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, SIGNUPS_ALLOWED, IP_HEADER

```json
{
  "_duo_akey": "***",
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": true,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": true,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://******.*********.*****",
  "domain_origin": "*****://******.*********.*****",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "invitation_org_name": "Bitwarden",
  "invitations_allowed": false,
  "ip_header": "CF-Connecting-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/bitwarden.log",
  "log_level": "debug",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": true,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "******@*********.*****",
  "smtp_from_name": "Bitwarden TobiASS",
  "smtp_host": "****.*****.**",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": "******@*********.*****",
  "templates_folder": "data/templates",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": "68256",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

### Steps to reproduce
Fill disk up to 99% on Linux.

### Expected behaviour
I expected to be still be able to login, but not being able to save passwords.

### Actual behaviour
When trying to login with my password i was rejected with wrong password. I got unauthenticated on any Android app or desktop app i was logged into prior to this incident. I got a white page while going to the admin panel --> users.

### Troubleshooting data
Copied this string from docker logging: `logging error: No space left on device (os error 28)Error performing logging` but forgot to take screenshots because i shit my pants during this incident.

**Comment by RealOrangeOne** [comment]
Systems write to disks for a variety of reasons. I've not dug into the database, but even writing session tokens to the database could do this. With a system without any free space, all bets are off in terms of stability, it doesn't just make it magically allow everything read only.

**Comment by BlackDex** [comment]
Device logins are saved to the database, login attempts are sent to the log or at least stdout/stderr also via Docker, which also uses disk storage. All these processes are touching the disk in some way.

As @RealOrangeOne already said, if the disk is full, a lot of items will get unstable are will produce issues, not only Vaultwarden. It is not up to Vaultwarden to keep a track of the disk-space left on the device.

**Reply by TobiasS1402** [reply]
Yeah I get that, and it was my own mistake. But I thought it could possibly be a bug.
